src/main/java/frc/robot/subsystems/climber/ClimberIOTalonFX.java

```diff
@@ -0,0 +1,304 @@
+package frc.robot.subsystems.climber;
+
+import static frc.robot.subsystems.climber.ClimberConstants.*;
+
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.StatusCode;
+import com.ctre.phoenix6.StatusSignal;
+import com.ctre.phoenix6.configs.CurrentLimitsConfigs;
+import com.ctre.phoenix6.configs.MotionMagicConfigs;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.MotionMagicExpoTorqueCurrentFOC;
+import com.ctre.phoenix6.controls.TorqueCurrentFOC;
+import com.ctre.phoenix6.controls.VelocityDutyCycle;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+
+import edu.wpi.first.units.Current;
+import frc.lib.team3015.subsystem.FaultReporter;
+import frc.lib.team3061.RobotConfig;
+import frc.lib.team3061.drivetrain.swerve.Conversions;
+import frc.lib.team6328.util.Alert;
+import frc.lib.team6328.util.Alert.AlertType;
+import frc.lib.team6328.util.TunableNumber;
+
+/** TalonFX implementation of the generic SubsystemIO */
+public class ClimberIOTalonFX implements ClimberIO {
+  private TalonFX leftMotor;
+  private TalonFX rightMotor;
+
+  private MotionMagicExpoTorqueCurrentFOC leftPositionCurrentRequest;
+  private TorqueCurrentFOC leftSupplyAmpsRequest;
+
+  private MotionMagicExpoTorqueCurrentFOC rightPositionCurrentRequest;
+  private TorqueCurrentFOC rightSupplyAmpsRequest;
+
+  private double leftRequestedReferenceSetpointMeters;
+  private double leftRequestedSupplyAmps;
+
+  private double rightRequestedReferenceSetpointMeters;
+  private double rightRequestedSupplyAmps;
+
+  private StatusSignal<Double> leftVelocityRPSStatusSignal;
+  private StatusSignal<Double> leftPositionMeterStatusSignal;
+  private StatusSignal<Double> leftStatorCurrentAmpsStatusSignal;
+  private StatusSignal<Double> leftSetpointStatusSignal;
+  private StatusSignal<Double> leftSupplyCurrentAmpsStatusSignal;
+
+  private StatusSignal<Double> rightVelocityRPSStatusSignal;
+  private StatusSignal<Double> rightPositionMetersStatusSignal;
+  private StatusSignal<Double> rightStatorCurrentAmpsStatusSignal;
+  private StatusSignal<Double> rightSetpointStatusSignal;
+  private StatusSignal<Double> rightSupplyCurrentAmpsStatusSignal;
+
+  private Alert configAlert =
+      new Alert("Failed to apply configuration for Climber.", AlertType.ERROR);
+
+  // FIXME: Add KV, KA, 
+  private final TunableNumber leftMotorKP =
+      new TunableNumber("Climber/leftMotorKP", LEFT_POSITION_PID_P);
```

Comment:
Just one set of Tunables for both left and right (see comment above about constants).